### DISCUSSION TITLE
Modpack Crash Fabric

### DISCUSSION THREAD

**Discussion by OxyV** [FabricMC/fabric-api#2631] [Player support] [answered]
Hello. 😢 

  Well i created a modpack from curseforge and i got many many issues when i tried to create a dedicated fabric server with the mods i've downloaded from curseforge.
 
I don't know much about creating a modpack for servers without crashing it, and i didn't want to make it worse. I got more than 200 mods (bruh)

 If you have time, please help me with the problems slowly. I don't want to take your time too much. I'm not asking you to solve all the problems at once. Because i think i got a lot of issues. (i guess)

I can delete some mods if you want me to. 

Minecraft Version : 1.19.2
Fabric Version : 0.14.10

Latest Crash: https://pastebin.pl/view/d142701f
Log: https://pastebin.pl/view/56502de5

I found some problems in the log. Copper hopper, Mega Parrots, Some dungeon mods i think. ( I tried to help, i just don't know what to do)

Thank you so much if you decided to help me.

**Comment by Juuxel** [comment, marked as answer]
Most of the issues are recipe file format issues etc which don't cause hard crashes, only log spam. The actual issue seems to be with opening the port for the server:

> [21:40:07] [Server thread/WARN]: **** FAILED TO BIND TO PORT!
> [21:40:07] [Server thread/WARN]: The exception was: java.net.BindException: Cannot assign requested address: bind
> [21:40:07] [Server thread/WARN]: Perhaps a server is already running on that port?

**Reply by OxyV** [reply]
Yeah i saw that in the first place. But i didn't think it was the only problem. I didn't change the IP. Is there any problem that should i fix?
i don't want it to effect the game. 



And thank you so much for your help.

**Reply by warjort** [reply]
Usually, it's because you have the server already running. As the WARNing suggests.

Here's a list of things to look at somebody wrote for the vanilla game:
https://www.partitionwizard.com/partitionmagic/minecraft-server-failed-to-bind-to-port.html
Server already running is number 1.

**Reply by OxyV** [reply]
No no, i know the IP problem I already fixed it.
 
 Well my grammar sucks sometimes. Sorry for that man.
 What i'm trying to say that is there any problems besides the IP?

**Reply by OxyV** [reply]
:(

**Reply by warjort** [reply]
If I understand you correctly, you have a different problem after you fixed the server address binding problem?
If that is true, you need to post your new log so we can see the error.

**Reply by OxyV** [reply]
Nope. I'm just saying that is there anything else that needs to be fixed in the log? I see a lot of warns and errors. Maybe they will make a issue when i play long term? 


OH i remembered a issue. I cannot change gamemode when i'm in the server.
![Screenshot_2](https://user-images.githubusercontent.com/117309771/200194003-13bdc9fd-8a72-4006-a2af-5fa39a4880d7.png)


Here is the text for the error:

Cannot invoke ''com.mojang.brigadier.CommandDispatcher.execute(String,Object)''because
''net.fabricmc.fabric.impl.command.client.ClientCommandInternals.activeDispatcher'' is null


Thank you.

**Reply by warjort** [reply]
The errors and warnings need to be referred to the relevant mod authors to see if there is a problem.
e.g. you have
````
[21:40:04] [main/ERROR]: Couldn't parse loot table mysticalcrops:blocks/tin_crop
java.lang.IllegalArgumentException: Can't find block mysticalcrops:tin_crop
````
so contact the mystical crops mod author.

For the command error you need to post the log so we can see the full error.
That snippet of an error doesn't give the full context needed to diagnose the problem.
If it did, you could probably understand and fix it yourself. :-)

**Reply by OxyV** [reply]
I did post the log up there. 

Thank you for your answer. There is a lot of errors so i think it will take a long time to contact xd

**Reply by warjort** [reply]
I don't see it. The only log is for your address binding problem.

Start a new discussion thread for this new topic and post a link to the log file there.
That way we can clearly see what is going on and are not mixing up different issues.

**Reply by OxyV** [reply]
I solved the problem. 

Well i typed in the minecraft server launcher thing. 
![Screenshot_3](https://user-images.githubusercontent.com/117309771/200198601-2026dd85-9de8-49d5-970f-5dd4657d10c2.png)


It worked. I think that the game itself have a protection or something. (Maybe because from a mod?) But they needed to be opped for cheats tho? 

Idk but i solved it. 

Thanks man. Take care.
